packages/sdk/src/sdk/services/EntityManager/EntityManagerClient.ts

```diff
@@ -74,40 +73,34 @@ export class EntityManager implements EntityManagerService {
     entityId,
     action,
     metadata = '',
-    auth,
     confirmationTimeout = CONFIRMATION_TIMEOUT,
     skipConfirmation = false
   }: ManageEntityOptions): Promise<
     Pick<TransactionReceipt, 'blockHash' | 'blockNumber'>
   > {
-    await this.init()
     const nonce = await getNonce()
-    const chainId = Number(this.config.chainId)
-    const signatureData = generators.getManageEntityData(
-      chainId,
-      this.config.contractAddress,
-      // TODO: This should be required, but not certain all
-      // callers are requiring it.
-      userId!,
-      entityType,
-      entityId,
-      action,
-      metadata,
-      nonce
-    )
 
-    const senderAddress = await auth.getAddress()
-    const signature = await auth.signTransaction(signatureData)
-
-    const method = await this.contract.methods.manageEntity(
-      userId,
-      entityType,
-      entityId,
-      action,
-      metadata,
-      nonce,
-      signature
-    )
+    const typedData: TypedDataDefinition<EntityManagerTypes, 'ManageEntity'> = {
+      domain: {
+        name: 'Entity Manager',
+        chainId: BigInt(this.chainId),
+        version: '1',
+        verifyingContract: this.contractAddress as Hex
+      },
+      primaryType: 'ManageEntity',
+      message: {
+        userId: userId!,
```

Comment:
I see you left out the comment about how we should type this better and check the callers 😅

Reply:
Good catch!